modules/contracts/src.ts/services/ethService.ts

```diff
@@ -684,6 +660,80 @@ export class EthereumChainService extends EthereumChainReader implements IVector
     }
   }
 
+  private async waitForConfirmation(
+    channelAddress: string,
+    chainId: number,
+    reason: TransactionReason,
+    response: TransactionResponse,
+  ): Promise<TransactionReceipt | undefined> {
+    // Register callbacks for saving tx, then return.
+
+    // An anon fn to get the tx receipt, as we may require multiple retries with raised gas price.
+    const getTransactionReceipt = async (): Promise<TransactionReceipt | undefined> => {
+      // TODO: This should be replaced with a polling method (?)
+      // TODO: If we're polling here, we shouldn't be awaiting a receipt, but checking to see if it's available
+      // (i.e. it's been confirmed / mined).
+      return await ethers.provider.getTransactionReceipt(response.hash);
+      // TODO: If the tx has not yet been mined, return undefined.
+      // return undefined;
+    }
+
+    let receipt: TransactionReceipt | undefined;
+    try {
+      // Poll for receipt.
+      receipt = await getTransactionReceipt();
+      // TODO: Should we have a diff. system of defaultRetries for this operation?
+      // NOTE: This loop won't execute if receipt is valid (not undefined).
+      for (let attempt = 1; attempt < this.defaultRetries && !receipt; attempt++) {
```

Comment:
Should not be related to `defaultRetries` which is related to tx sending. I think this basically needs to be an infinite loop that keeps bumping gas. Maybe the ultimate timeout is when it gets to the MAX_GAS_PRICE we configure.

Reply:
Sounds good. TODO is left there, so safe to say I will handle that.

I'll go with a `while` loop and use gas price and !receipt to handle this